File: practice/mathematics/algebra/stepping_stones_game/stepping_stones_game.cpp
#include <iostream>
#include <cmath>

using namespace std;

int main()
{
    int t;
    long long n;
    
    cin >> t;
    while (t--)
    {
        cin >> n;
        long long k = (long long)sqrt(2 * n);
        while (k * (k + 1) < 2 * n)
        {
            k++;
        }
        if (k * (k + 1) == 2 * n)
        {
            cout << "Go On Bob " << k << endl;
        }
        else
        {
            cout << "Better Luck Next Time" << endl;   
        }
    }
    
    return 0;
}

File: practice/mathematics/algebra/stepping_stones_game/stepping_stones_game.hs
module Main where

import Control.Monad (forM_)

isqrt :: Integer -> Integer
isqrt = floor . sqrt . fromInteger

solve k n
    | k * (k + 1) < 2 * n  = solve (k + 1) n
    | k * (k + 1) == 2 * n = "Go On Bob " ++ (show k)
    | otherwise            = "Better Luck Next Time"

main :: IO ()
main = do
    t <- readLn :: IO Int
    forM_ [1..t] $ \_ -> do
        n <- readLn :: IO Integer
        let k = isqrt (2 * n)
        putStrLn $ solve k n

